integration-cli/docker_cli_external_volume_driver_unix_test.go

```diff
@@ -616,3 +616,18 @@ func (s *DockerExternalVolumeSuite) TestExternalVolumeDriverUnmountOnMountFail(c
 	out, _ = s.d.Cmd("run", "-w", "/foo", "-v", "testumount:/foo", "busybox", "true")
 	c.Assert(s.ec.unmounts, checker.Equals, 0, check.Commentf(out))
 }
+
+func (s *DockerExternalVolumeSuite) TestExternalVolumeDriverUnmountOnCp(c *check.C) {
+	s.d.StartWithBusybox(c)
+	s.d.Cmd("volume", "create", "-d", "test-external-volume-driver", "--name=test")
+
+	out, _ := s.d.Cmd("run", "-d", "--name=test", "-v", "test:/foo", "busybox", "/bin/sh", "-c", "touch /test && top")
```

Comment:
Using the new cli framework will give you much better errors here when something fails, making it a lot easier to debug:

```go
result := cli.Docker(c, cli.Args("run", "-d", ...), cli.Daemon(s.d))
result.Assert(c, icmd.Success)
c.Assert(s.ec.mounts, checker.Equals, 1, check.Commentf(result.Combined()))
```